src/Client.php

```diff
@@ -99,6 +99,20 @@ public function __construct($uri = 'mongodb://127.0.0.1/', array $uriOptions = [
             throw InvalidArgumentException::invalidType('"typeMap" driver option', $driverOptions['typeMap'], 'array');
         }
 
+        if (isset($driverOptions['autoEncryption'])) {
+            if (! is_array($driverOptions['autoEncryption'])) {
+                throw InvalidArgumentException::invalidType('"autoEncryption" driver option', $driverOptions['autoEncryption'], 'array');
+            }
+
+            if (isset($driverOptions['autoEncryption']['keyVaultClient'])) {
+                if (! $driverOptions['autoEncryption']['keyVaultClient'] instanceof self) {
```

Comment:
I wonder if we can just reduce the entire block of code added to this constructor to:

```php
if (isset($driverOptions['autoEncryption']['keyVaultClient']) &&
    $driverOptions['autoEncryption']['keyVaultClient'] instanceof self) {
    $driverOptions['autoEncryption']['keyVaultClient'] = $driverOptions['autoEncryption']['keyVaultClient']->manager;
}
```

IIRC, PHPC will already check that `autoEncryption` is an array and go on to check that `keyVaultClient` is a Manager instance. The only responsibility of PHPLIB would be to "unwrap" a Client instance into a Manager. I also don't think we need to prohibit folks from passing a Manager here.

I'll note that the [`Client::__construct()`](https://docs.mongodb.com/php-library/current/reference/method/MongoDBClient__construct/#definition) docs originally didn't go into detail about driver-specific options (i.e. everything other than `typeMap`), although we do appear to duplicate those docs in the current text. If users were to go right to the PHPC docs for information on driver options, they might actually think they need to pass a Manager instance. I suppose this PR should also revise the documentation to add an entry for `autoEncryption` with a brief description that (a) links to the PHPC docs and (b) notes that users can pass a `MongoDB\Client` in lieu of a `MongoDB\Driver\Manager` if they wish, which the library will unwrap for them.

Side note: I just realized we missed deprecating SSL driver options -- I've opened [PHPLIB-521](https://jira.mongodb.org/browse/PHPLIB-521) to track that.

Reply:
> I wonder if we can just reduce the entire block of code added to this constructor to:

Technically, yes. However, that changes the exception message to be "Expected "keyVaultClient" encryption option to be MongoDB\Driver\Manager, string given", which can be misleading as we allow passing a `MongoDB\Client` as well.

I can change the code to check if `$driverOptions['autoEncryption']['keyVaultClient']` is set, handle unwrapping if necessary and throw only if the `keyVaultClient` is not an instance of `MongoDB\Client` or `MongoDB\Driver\Manager`. As is, we disallow passing a manager directly, which may still be a valid use-case for some people.